10up/restricted-site-access
# Add: implement custom https header based whitelisting

After team discussion, we decided to use `restricted_site_access_is_restricted` filter hook to provide a code snippet to implement the same feature and add that to the README file.

-----------


### Description of the Change



I add `rsa_custom_trusted_headers` filter hook to allow custom headers with unique values.
Example:

```php
add_filter( 'rsa_custom_trusted_headers', 'my_rsa_custom_trusted_headers' );
function my_rsa_custom_trusted_headers ( $headers ) {
  $headers['x-gt-matrix'] = 'SomeValue1';
  $headers['x-custom-header'] = 'SomeValue2';
  
  return $headers;
}
```

Closes #179 

### How to test the Change

- Allow custom headers as mentioned in the code snippet.
- Use tools like postmen to request URLs with custom headers.
- Check whether or not you can view a restricted site.

### Changelog Entry

> Added - implement custom trusted header based whitelisting


### Credits

Props @dkotter , @jeffpaul 


### Checklist:
